doc/manual/chapters/translation.tex

```diff
@@ -340,118 +684,124 @@ \subsection{With}
 
 \noindent
 \begin{minipage}[t]{\textwidth}
 \begin{lstlisting}[language=golang]
-import (
-	"math/rand"
-	"pgo/datatypes"
-	"time"
-)
-
-var S_1 datatypes.Set = datatypes.NewSet(1, 2, 3)
-var a string = "foo"
-// ...
-func main() {
-	rand.Seed(time.Now().UnixNano())
-	// ...
-	{
-		var x_1 int = S_1.ToSlice()[rand.Intn(S_1.Size())].(int)
-		var x_2 string = a
-		// ...
-		// do stuff with the x_i ...
-	}
-}
+S_1 := []int{1, 2, 3}
+a := "foo"
+x_1 := S_1[0]
+x_2 := a
 \end{lstlisting}
 \captionof{figure}{Compiled Go}
 \end{minipage}
 
-In Go the random number generator is seeded with the current Unix time at the beginning of the \texttt{main} function. The local variables declared by the \texttt{with} and its body are contained in a separate code block, to preserve the variables' local scope.
+The local variables declared by the \texttt{with} and its body are potentially renamed to ensure no accidental capture.
 
 \subsection{Processes}
-The PlusCal algorithm body can either contain statements in a uniprocess algorithm, or process declarations in a multiprocess algorithm. Processes can be declared with the syntax \verb|process (Name \in S)| or \verb|process (Name = Id)|. The first construct spawns a set of processes, one for each ID in the set \texttt{S}, and the second spawns a single process with ID \texttt{Id}. A process can refer to its identifier with the keyword \texttt{self}. In TLC, multiprocess algorithms are simulated by repeatedly selecting a random process then advancing it one atomic step (if it does not block).
+The PlusCal algorithm body can either contain statements in a uniprocess algorithm, or process declarations in a multiprocess algorithm.
+
+Uniprocess algorithms are translated into single-threaded Go programs.
 
-PGo converts each process body to a function and spawns a goroutine per process. The function takes a single parameter \texttt{self}, the process ID. There are two semantic considerations: the main goroutine should not exit before all goroutines finish executing, and the time at which all child goroutines begin executing should be synchronized. To preserve these semantics, PGo uses a global waitgroup which waits on all goroutines, and each process body pulls from a dummy channel before beginning execution. When all goroutines have been initialized, the dummy channel is closed so that the channel pull no longer blocks, allowing for a synchronized start.
+In a multiprocess algorithm, processes can be declared with the syntax \verb|process (Name \in S)| or \verb|process (Name = Id)|. The first construct spawns a set of processes, one for each ID in the set \texttt{S}, and the second spawns a single process with ID \texttt{Id}. A process can refer to its identifier with the keyword \texttt{self}. In TLC, multiprocess algorithms are simulated by repeatedly selecting a random process then advancing it one atomic step (if it does not block).
```

Comment:
> A process can refer to its identifier with the keyword \texttt{self}. In TLC, multiprocess algorithms are simulated by repeatedly selecting a random process then advancing it one atomic step (if it does not block).

I'd not go into the details of how TLC handles model checking since this is mostly orthogonal to PGo. That said, if you prefer to have such sentence, I wouldn't use the word "blocking", and instead talk about exploration of enabled states, once again to use wording that is common in TLA+ documents (assuming the reader is familiar with those).